### ISSUE TITLE
Roadmap to a 8.x stable release

### ISSUE THREAD

**Issue by zerolab** [Gizra/og#240]
Hi,

We have decided to go with OG for a new D8 project. While there are some rough edges, it would be great to get a consensus about the roadmap to a stable release and what that will entail. 

Adding a *very* rough outline from existing PRs and issues to get the conversation started. Please feel free to update the list or, at least provide some guidance.

Thank you for the hard work!

---

## Alpha
- https://github.com/Gizra/og/releases/tag/8.x-1.0-alpha6

## Beta

- #181 (Remove OG complex widget)
- #242 (OG Access port)
- #243 (Group roles and permissions UI)
- #237 (Private group functionality)

## Stable

- #432 (Migration path from D6/D7)

**Comment by kentr** [comment]
@zerolab How "rough" are the rough edges?  Is OG usable for D8?

**Comment by zerolab** [comment]
The PRs above bring it to a much more polished state.
It is already use in production by https://github.com/ec-europa/joinup-dev. Our build is in progress.

With a bit more work it will be in good shape. Testing and reviewing existing PRs would help. Additional changes are very welcome (and I speak as a user of the module)

**Comment by pfrenssen** [comment]
@kentr if you decide to use a dev build (or even an alpha build) the API is not considered to be stable, so it is not recommended for production. Any new commit on the dev build, or new alpha releases might break existing implementations.

That said, it is a matter of making a risk / benefit analysis for your own particular situation. For our project we are migrating a very large D6 website that uses OG. We consider the risk for API breakage (i.e. the time needed to refactor our code base when the API changes) much lower than the cost of having to invent our own solution, or moving to a different solution. We are mitigating the risk of being confronted with undiscovered security issues or critical bugs by having our own very extensive test suite.

So if OG is important for your project you can use it, but be prepared to do additional testing and have a potentially higher maintenance cost until it stabilizes.

**Comment by MPParsley** [comment]
I agree with @pfrenssen that providing your own test suites (e.g. Behat functional tests) are a must to mitigate these issues. Especially the use case for protected content makes OG prone to security issues (e.g. information disclosure).

The OG module and it's API's are quite well covered by unit and kernel tests, but as long as it's dev those might still break so it would be nice to have a more visible roadmap towards a stable release.

The label Alpha Blocker seems to give an indication https://github.com/Gizra/og/labels/Alpha%20blocker

**Comment by jerrac** [comment]
Hey all, 

Can we get an update on the status of OG for Drupal 8?

I've been poking around and watching issues and pull requests for the past couple months. Haven't found anywhere I thought I could help, but I am eager to see a D8 release. I have a project waiting for it. (I actually floated the idea of paying Gizra to finish things up past my boss, but the project isn't something we'd be able to find money for with our budget where it is. :( )

It looks like there are several pull requests that are almost ready to be merged, and only 6 issues have both the "beta blocker" and "drupal 8" labels. So it feels like things are really close. 

Thanks!

**Comment by pfrenssen** [comment]
We're currently down to 6 issues before we plan to release a beta version: https://github.com/Gizra/og/issues?q=is%3Aissue+is%3Aopen+label%3A%22Beta+blocker%22

Any help to finish these is more than welcome!

**Comment by yookoala** [comment]
Any update on the beta release yet?

**Comment by ericveal** [comment]
Any update on the beta release? Thanks!
